audio-worklet/src/noisegate-script-processor.js

```diff
@@ -0,0 +1,195 @@
+/**
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+class NoiseGate {
+  /**
+   * A Noise gate allows audio signals to pass only when the registered volume
+   * is above a specified threshold.
+   * @param {BaseAudioContext} context the audio context
+   * @param {Number} options.channels the number of input and output
+   *                                  channels, a maximum of two
+   * @param {Object} options parameters for the noise gate
+   * @param {Number} options.attack seconds for gate to fully close
+   * @param {Number} options.release seconds for gate to fully open
+   * @param {Number} options.bufferSize the size of an onaudioprocess window
+   * @param {Number} options.timeConstant seconds for envelope follower's
+   *                                   smoothing filter alpha weight
+   * @param {Number} options.threshold decibel level beneath which sound is
+   *                                   muted
+   */
+  constructor(context, options) {
+    if (!(context instanceof BaseAudioContext))
+      throw 'Not a valid audio context.';
+    if (!options) options = {};
+    
+    const numberOfChannels = options.numberOfChannels || 1;
+    if (numberOfChannels > 2)
+      throw 'The maximum supported number of channels is two.';
+
+    this.context_ = context;
+    const bufferSize = options.bufferSize || 0;
+    // The default threshold is 140 decibels (loudest sound a human can hear
+    // without pain) so that the noise gate does not mute signals by default.
+    this.threshold = options.threshold || 140;
```

Comment:
Well, this is really weird. The threshold in the dynamic processors is commonly in dBFS and the noise gate's threshold is usually between -100~0 dBFS. This is a simple noise gate from Logic:

![image](https://user-images.githubusercontent.com/676891/28236679-e7b60c32-68e0-11e7-9c71-7cc6182c15ad.png)